epochStart/shardchain/trigger.go

```diff
@@ -1080,6 +1093,124 @@ func (t *trigger) RevertStateToBlock(header data.HeaderHandler) error {
 	return nil
 }
 
+// this has to be run under trigger mutex
+func (t *trigger) saveEpochStartInfoToStaticStorer() error {
+	epoch := t.metaEpoch
+
+	epochStartMetablock, err := t.saveEpochStartBlockToStaticStorer(epoch)
+	if err != nil {
+		log.Debug("saveEpochStartBlockToStaticStorage failed", "error", err.Error())
+		return err
+	}
+
+	err = t.savePeerMiniBlocksToStaticStorer(epochStartMetablock)
+	if err != nil {
+		log.Debug("savePeerMiniBlocksToStaticStorer failed", "error", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (t *trigger) saveEpochStartBlockToStaticStorer(epoch uint32) (data.HeaderHandler, error) {
+	epochStartIdentifier := core.EpochStartIdentifier(epoch)
+	marshalledMetaBlock, err := t.metaHdrStorage.Get([]byte(epochStartIdentifier))
+	if err != nil {
+		return nil, err
+	}
+
+	epochStartMetablock := &block.MetaBlock{}
+	err = t.marshaller.Unmarshal(epochStartMetablock, marshalledMetaBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	epochStartBootstrapKey := append([]byte(common.EpochStartStaticBootstrapKeyPrefix), []byte(fmt.Sprint(epoch))...)
+	err = t.epochStartStaticStorage.Put(epochStartBootstrapKey, marshalledMetaBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debug("saveEpochStartInfoToStaticStorage put metaBlock into epochStartStaticStorage", "epoch", epoch) // TODO: remove log
```

Comment:
I think the log can stay, it should appear once per epoch

Reply:
:+1: